### ISSUE TITLE
Consider adding filter functionality

### ISSUE THREAD

**Issue by 9ktz** [TheBlob42/drex.nvim#27]
Hey, not sure if that's out of your intended scope but its certainly a feature I'd make a lot of use of.

**Comment by TheBlob42** [comment]
Could you elaborate in more detail what you mean by "live filtering"? :thinking: 
And also some example use cases how you would use it, so that I can check if _A)_ there is already a solution for it and _B)_ better understand what you're aiming for here and how it could be potentially be implemented

**Comment by TheBlob42** [comment]
I actually work on a "live search" feature to better work with drex, so maybe this is exactly what you're looking for :slightly_smiling_face:  will notify you as soon as it hits the `develop` branch so you could test it

I will note down your suggestions, but I can't make any promise if and when I'll find the time to have a look at them

#### Some background
Drex was born as a very personal project to solve the issues that I faced with other alternatives at the time I started. And for me it is pretty feature complete. I use it everyday at work and I am quite happy with its current state. I will work on it whenever I see a bug, have a good idea for a new feature or if somebody reports an issue or submits a PR here on Github. But I have no plan (and especially not the time) to make it feature equivalent to any of those more "mature" alternatives like `neo-tree`, `nvim-tree`, `ranger`, `nnn`, etc. in the near future. Maybe it'll get there at some point, but there is no rush for it

And for everybody else reading this: If you see a feature that you would like to see in drex go ahead and submit a PR for it. I am always happy to help you out if you have problems making it work or would love some feedback :grinning:

**Comment by 9ktz** [comment]
That's definitely nice, a search of any sorts is definitely needed. In fact I should title the issue search. I named it filter after nvim-tree which calls it like this, which I happened to reside in the meantime as I assumed bit of the information written in the background section by intuition.

I'll be checking on drex here and there and maybe switch back to it some day or another. Thanks for your reply and time :)
